### PR TITLE
fix invalid invocation on getRandomValues()

### PR DESCRIPTION
**Pull request by anonrig**
Fixes https://github.com/cloudflare/workerd/issues/2716

### CODE REVIEW

File: src/workerd/api/node/tests/crypto_random-test.js [hunk 1]
```diff
@@ -413,3 +413,12 @@ export const timingSafeEqualTest = {
     timingSafeEqual(new Uint8Array(1), new Uint8Array(1));
   },
 };
+
+// Ref: https://github.com/cloudflare/workerd/issues/2716
+export const getRandomValuesIllegalInvocation = {
+  async test() {
+    const crypto = await import('node:crypto');
+    const getRandomValues = crypto.getRandomValues;
```
Comment (petebacondarwin) [inline src/workerd/api/node/tests/crypto_random-test.js:421]: Can you also check `const getRandomValues = crypto.webcrypto.getRandomValues`?

I think that this still is not able to be used unbound.

**Review by petebacondarwin — COMMENTED** [review]
Thanks for fixing this. Can you check that other scenario too?

File: src/workerd/api/node/tests/crypto_random-test.js (hunk 1 above)
Comment (petebacondarwin) [reply src/workerd/api/node/tests/crypto_random-test.js]: Gosh! I was sure I tested that previously but I just tried now and I can confirm that this case does indeed throw.
Thanks!

File: src/workerd/api/node/tests/crypto_random-test.js [hunk 2]
```diff
@@ -413,3 +413,27 @@ export const timingSafeEqualTest = {
     timingSafeEqual(new Uint8Array(1), new Uint8Array(1));
   },
 };
+
+// Ref: https://github.com/cloudflare/workerd/issues/2716
+export const getRandomValuesIllegalInvocation = {
+  async test() {
+    const crypto = await import('node:crypto');
+    {
+      // The following assertion doesn't fail as of Node.js v20.17.0
```
Comment (petebacondarwin) [inline src/workerd/api/node/tests/crypto_random-test.js:422]: This doesn't fail for me in node 18 either...

**Review by petebacondarwin — APPROVED** [review]

**Comment by IgorMinar** [comment]
thank you @anonrig!!

**Comment by anonrig** [comment]
@IgorMinar 🫡

**Review by dom96 — APPROVED** [review]

**Pull request merged**